pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/VfHandler.java

```diff
@@ -28,6 +29,23 @@ public Parser getParser(ParserOptions parserOptions) {
         return new VfParser(parserOptions);
     }
 
+    @Override
+    public VisitorStarter getTypeResolutionFacade(ClassLoader classLoader) {
```

Comment:
This pattern seemed like the correct way to invoke the visitor. I hadn't noticed it in my previous commit.

Reply:
This is technically the way but it's deprecated for removal. In PMD 7 we'll centralise the logic for such passes in the implementation of the `Parser#parse` routine. Conveniently, the parse routine is also the place where you have access to a ParserOptions. So you can avoid putting them on the ASTCompilationUnit as public data.

I suggest you instead move this logic into `VfParser#parse` and remove `ASTCompilationUnit#getPropertySource`